agent/config/parse.go

```diff
@@ -27,6 +27,13 @@ import (
 	cnitypes "github.com/containernetworking/cni/pkg/types"
 )
 
+const (
+	// envSkipDomainJoinCheck is an environment setting that can be used to skip
+	// domain join check validation. This is useful for integration and
+	// functional-tests but should not be set for any non-test use-case.
+	envSkipDomainJoinCheck = "ZZZ_SKIP_DOMAIN_JOIN_CHECK_NOT_SUPPORTED_IN_PRODUCTION"
```

Comment:
nit do we need the `ZZZ`?

Reply:
this flag is used by windows gMSA as well, It is possible i might break stuff if I change it. I am reusing existing one